src/backends/pgfplots.jl

```diff
@@ -10,6 +9,134 @@ function _initialize_backend(::PGFPlotsBackend; kw...)
   # TODO: other initialization
 end
 
+_pgfplots_get_color(c) = "{rgb,1:red,$(float(c.r));green,$(float(c.g));blue,$(float(c.b))}"
+
+function _pgfplots_get_linestyle!(kwargs, plt)
+    linestyle = plt[:linestyle]
+    if linestyle == :dash
+        kwargs[:style] *= "dashed,"
+    elseif linestyle == :dot
+        kwargs[:style] *= "dotted,"
+    elseif linestyle == :dashdot
+        kwargs[:style] *= "dashdotted,"
+    elseif linestyle == :dashdotdot
+        kwargs[:style] *= "dashdotdotted,"
+    end
+
+    kwargs[:style] *= (haskey(plt, :linewidth) && plt[:linewidth] != :auto) ? "line width = $(plt[:linewidth]) pt," : "line width = 1 pt,"
+end
+
+
+function _pgfplots_get_marker!(kwargs, plt)
+    # Control marker shape
+    mark = plt[:markershape]
+    if mark in (:none,:n, :no)
+        kwargs[:style] *= "mark = none,"
+    elseif mark in (:auto, :a)
```

Comment:
Yes I do. For lims we need to know to not change the defaults, whereas auto for a marker means we need to select a shape based on the series index, so it gets replaced with an actual shape. 

> On Apr 28, 2016, at 7:43 AM, Patrick Kofod Mogensen <notifications@github.com> wrote:
> 
> In src/backends/pgfplots.jl:
> 
> > +    elseif linestyle == :dashdot
> > +        kwargs[:style] *= "dashdotted,"
> > +    elseif linestyle == :dashdotdot
> > +        kwargs[:style] *= "dashdotdotted,"
> > +    end
> > +
> > +    kwargs[:style] *= (haskey(plt, :linewidth) && plt[:linewidth] != :auto) ? "line width = $(plt[:linewidth]) pt," : "line width = 1 pt,"
> > +end
> > +
> > +
> > +function _pgfplots_get_marker!(kwargs, plt)
> > +    # Control marker shape
> > +    mark = plt[:markershape]
> > +    if mark in (:none,:n, :no)
> > +        kwargs[:style] *= "mark = none,"
> > +    elseif mark in (:auto, :a)
> Do you mean for marks only? It does show up in :xlims and :ylims (and :zlims I guess).
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
>